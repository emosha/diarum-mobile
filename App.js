import React from 'react';
import ApolloClient from 'apollo-client';
import { ApolloProvider } from 'react-apollo';
import { ApolloLink } from 'apollo-link';
import { createHttpLink } from 'apollo-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { setContext } from 'apollo-link-context';

import AppNavigator from './components/AppNavigator';
import DropdownAlert from './components/DropdownAlert';

const httpLink = createHttpLink({
  uri: process.env.BASE_URL,
});

const authLink = setContext((_, { headers }) => ({
  headers: {
    ...headers,
    authorization: process.env.JWT,
  },
}));

export const client = new ApolloClient({
  link: ApolloLink.from([authLink, httpLink]),
  cache: new InMemoryCache(),
});

export default class App extends React.Component {
  state = {
    showAlert: false,
    options: {},
  }

  /**
   * Shows dropdown alert
   *
   * @returns {void}
   */
  showDropdownAlert = (showAlert, options) => {
    this.setState({
      showAlert,
      options,
    });
  }

  render() {
    return (
      <ApolloProvider client={client}>
        <AppNavigator screenProps={{ showDropdownAlert: this.showDropdownAlert }} />
        <DropdownAlert
          showAlert={this.state.showAlert}
          options={this.state.options}
        />
      </ApolloProvider>
    );
  }
}
