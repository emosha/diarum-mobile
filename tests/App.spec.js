import React from 'react';
import App from '../App';

describe('App', () => {
  let Component;
  let instance;

  beforeAll(() => {
    Component = mount(<App />);
    instance = Component.instance();
  });

  it('renders correctly', () => {
    expect(Component).toMatchSnapshot();
  });

  it('sets showAlert and options values in state when showDropdownAlert method is called', () => {
    const options = {
      type: 'error',
      title: 'Error',
      message: 'an error',
    };

    expect(instance.state).toEqual({
      showAlert: false,
      options: {},
    });

    instance.showDropdownAlert(true, options);

    expect(instance.state).toEqual({
      options,
      showAlert: true,
    });
  });
});
