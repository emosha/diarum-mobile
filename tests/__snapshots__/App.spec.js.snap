// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#5D59CB"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
